brickstrap-utils.sh

```diff
@@ -183,27 +196,25 @@ Do you have necessary binfmt packages installed?"
 #
 # Check whether commandline or environment based configuration of QEMU is
 # 'valid', i.e that it corresponds to an existing QEMU interpreter.
-# Returns 1 if no QEMU was set, to signal brp_determine_qemu
-# Fails if QEMU is explicitly misconfigured.
+# The variable 'BRP_BINFMT' is set to the passed QEMU interpreter if valid.
+# Returns 255 if no QEMU was configured. Fails if the QEMU settings are
+# misconfigured.
 #
 function brp_validate_qemu()
 {
     # check environment variable first, may have been set by config
     if [ -n "$QEMU_STATIC" -a -x "$QEMU_STATIC" ]; then
-        echo -n "$(readlink -f $QEMU_STATIC)"
+        BRP_BINFMT="$(readlink -f $QEMU_STATIC)"
```

Comment:
The distinctions are as follows:
 * `BR_QEMU` is namespaced as brickstrap+public, and it holds the raw `-Q` value. Project configurations may inspect this value but may not alter it. Because of the namespacing it's also not really valid to set this in the environment prior to invoking brickstrap.
 * `BRP_BINFMT` corresponds to the eventually determined binfmt binary in the *host* file system. This is not necessarily the same when viewed from inside the chroot. Examples of where this non-obvious detail matters: developer QEMU builds that live in home directories or `/usr/local/bin`. Hence, the copy logic contains comments to that effect. `BRP_BINFMT` is namespaced as brickstrap+private, so project configuration should not examine its value and in fact it is meant to be local to the `-utils.sh` module. I'll see about introducing a `brp_` namespaced getter, as a side effect of that it should be possible to also fix `create-tar` properly, again.
 * `QEMU_STATIC` is a globally namespaced variable which the project configuration may inspect *and* alter as it sees fit. Eventually it is set to the QEMU binary in the `rootfs`, as seen from inside the chroot.